xrpl/utils/txn_parser/utils/balance_parser.py

```diff
@@ -0,0 +1,184 @@
+                balance=Balance(
+                    currency="XRP",
+                    value=f"{xrp_value.normalize():f}",
+                ),
+            )
+    return None
+
+
+def _flip_trustline_perspective(balance_change: BalanceChange) -> BalanceChange:
+    balance = balance_change["balance"]
+    negated_value = Decimal(balance["value"]).copy_negate()
+    issuer = balance["issuer"]
+    return BalanceChange(
+        account=issuer,
+        balance=Balance(
+            currency=balance["currency"],
+            issuer=balance_change["account"],
+            value=f"{negated_value.normalize():f}",
+        ),
+    )
+
+
+def _get_trustline_quantity(
+    node: NormalizedNode,
+    value: Optional[Decimal],
+) -> List[BalanceChange]:
+    """
+    Computes the complete list of every balance that changed in the ledger
+    as a result of the given transaction.
+
+    Args:
+        node: The affected node.
+        value: The currency amount value.
+
+    Returns:
+        A list of balance changes.
+    """
+    if value is None:
+        return []
+    fields = (
+        node["NewFields"] if node.get("NewFields") is not None else node["FinalFields"]
+    )
+    assert fields is not None
+    low_limit = fields.get("LowLimit")
+    balance = fields.get("Balance")
+    high_limit = fields.get("HighLimit")
+    if low_limit is not None and balance is not None and high_limit is not None:
+        low_limit_issuer = low_limit.get("issuer")
+        assert isinstance(balance, dict)
+        balance_currency = balance.get("currency")
+        high_limit_issuer = high_limit.get("issuer")
+        if (
+            low_limit_issuer is not None
+            and balance_currency is not None
+            and high_limit_issuer is not None
+        ):
+            result = BalanceChange(
+                account=low_limit_issuer,
+                balance=Balance(
+                    currency=balance_currency,
+                    issuer=high_limit_issuer,
+                    value=f"{value.normalize():f}",
+                ),
+            )
+            return [result, _flip_trustline_perspective(result)]
+    return []
+
+
+def _group_balance_changes(
+    balance_changes: List[BalanceChange],
+) -> Dict[str, List[BalanceChange]]:
+    grouped_balance_changes: Dict[str, List[BalanceChange]] = {}
+    for change in balance_changes:
+        account = change["account"]
+        if account not in grouped_balance_changes:
+            grouped_balance_changes[account] = []
+        grouped_balance_changes[account].append(change)
+    return grouped_balance_changes
+
+
+def get_value(balance: Union[Dict[str, str], str]) -> Decimal:
+    """
+    Get a currency amount's value.
+
+    Args:
+        balance: Account's balance.
+
+    Returns:
+        The currency amount's value.
+    """
+    if isinstance(balance, str):
+        return Decimal(balance)
+    return Decimal(balance["value"])
+
+
+def get_quantities(
```

Comment:
A more descriptive name for this method might be `get_balance_changes` or `get_node_balance_changes`